alf/experience_replayers/replay_buffer.py

```diff
@@ -181,37 +249,35 @@ def get_batch(self, batch_size, batch_length):
                 self._flattened_buffer)
         return _convert_device(result)
 
-    def clear(self):
-        """Clear the replay bufer."""
-        self._current_size.fill_(0)
-        self._current_pos.fill_(0)
-
     def gather_all(self):
-        """Returns all the items in buffer.
+        """Returns all the items in the buffer.
 
         Returns:
-            Returns all the items currently in the buffer. The shapes of the
-            tensors are [B, T, ...] where B=num_environments, T=current_size
+            Tensors of shape [B, T, ...], B=num_environments, T=current_size
         Raises:
             AssertionError: if the current_size is not same for
                 all the environments
         """
         size = self._current_size.min()
         max_size = self._current_size.max()
         assert size == max_size, (
-            "Not all environment have the same size. min_size: %s "
+            "Not all environments have the same size. min_size: %s "
             "max_size: %s" % (size, max_size))
 
+        # NOTE: this is not the proper way to gather all from a ring
+        # buffer whose data can start from the middle, so this is limited
+        # to the case where clear() is the only way to remove data from
+        # the buffer.
```

Comment:
Perhaps you can assert _current_pos is 0?

Reply:
current_pos is the ending position of the data in the buffer, not the starting position, hence, current_pos == current_size when size < max_length.

Reply:
Right. That means we should assert current_size == current_pos?